File: app/src/main/java/com/yiche/example/newestversiondemo/fragment/TestPlatformActivityFragment.java
package com.yiche.example.newestversiondemo.fragment;

import android.os.Build;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.yiche.example.newestversiondemo.R;

/**
 * A placeholder fragment containing a simple view.
 */
public class TestPlatformActivityFragment extends Fragment {

    public TestPlatformActivityFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_test_platform, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        view.findViewById(R.id.underline).setVisibility(
                Build.VERSION.SDK_INT>=Build.VERSION_CODES.LOLLIPOP?View.GONE:View.VISIBLE);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }
}

File: app/src/main/java/com/yiche/example/newestversiondemo/fragment/TestPlatformActivityFragment.kt
package com.yiche.example.newestversiondemo.fragment

import android.content.Context
import android.os.Build
import android.support.v4.app.Fragment
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.yiche.example.newestversiondemo.R

/**
 * A placeholder fragment containing a simple view.
 */
class TestPlatformActivityFragment : Fragment() {

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        // 为啥把activity换成context
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d("Victor", "context: $context")
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_test_platform, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        view.findViewById<View>(R.id.underline).visibility = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) View.GONE else View.VISIBLE
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        Log.d("Victor", "activity: $activity")
    }
}
